src/js/components/accordion.js

```diff
@@ -0,0 +1,270 @@
+  * @param {string} selector - an element selector
+  * @param {Function} callback - a function the user wishes to trigger if the
+  * selector is below the maximum allowed number
+  * @param {number} max - the maximum number of a given selector allowed
+  */
+  _quantifySelector(entity, selector, callback, max = 1) {
+    let excess;
+
+    entity.querySelectorAll(selector).length > max ? excess = true : excess = false;
+
+    if (excess === true) {
+      throw new TypeError('Caught');
+    }
+
+    callback();
+  }
+
+  /**
+  * This function is used to trigger the 'accordionOpened' custom event, and
+  * open an accordion/panel pair. It removes the panel's hidden attribute,
+  * and sets the trigger's aria-expanded attribute to true. It also allows
+  * developers to setup a custom callback function.
+  * @param {Function} callback 
+  */
+  open(accordion, callback) {
+    // Trigger accordionOpened custom event. This event is used open accordion panels.
+
+    const activationEvent = dispatchCustomEvent(
+      'accordionOpened',
+      accordion,
+      {
+        id: accordion.dataset.accordionPanel,
+      }
+    );
+
+    if (!activationEvent) return;
+
+    const trigger = this.element.querySelector(`[${this.triggerAttribute} = "${accordion.dataset.accordionPanel}"]`);
+
+    // Open the appropriate accordion trigger/panel pair
+    accordion.removeAttribute('hidden');
+    trigger.setAttribute('aria-expanded', 'true');
+
+    if (callback && typeof callback === 'function') {
+      callback();
+    }
+  }
+
+  /**
+  * This function is used to trigger the 'accordionClosed' custom event, and
+  * close an accordion/panel pair. It adds the hidden attribute to the panel,
+  * and sets the trigger's aria-expanded attribute to false. It also allows
+  * developers to setup a custom callback function.
+  * @param {Function} callback 
+  */
+  close(accordion, callback) {
+    // Trigger accordionClosed custom event. This event is used to control the process of closing accordion panels.
+
+    const activationEvent = dispatchCustomEvent(
+      'accordionClosed',
+      accordion,
+      {
+        id: accordion.dataset.accordionPanel,
+      }
+    );
+
+    if (!activationEvent) return;
+
+    const trigger = this.element.querySelector(`[${this.triggerAttribute} = "${accordion.dataset.accordionPanel}"]`);
+
+    // Close the appropriate accordion trigger/panel pair
+    accordion.setAttribute('hidden', '');
+    trigger.setAttribute('aria-expanded', 'false');
+
+    if (callback && typeof callback === 'function') {
+      callback();
+    }
+  }
+
+  init() {
+    this._openOnInit();
+    this.destroy();
+
+    // Add click handlers
+    document.addEventListener('click', this._handleClick, false);
+    document.addEventListener('keydown', this._handleKeydown, false);
+  }
+
+  destroy() {
+    document.removeEventListener('click', this._handleClick, false);
+    document.removeEventListener('keydown', this._handleKeydown, false);
```

Comment:
These event listeners should be called on `this.element` instead of `document`. Right now if you add more than one accordion to the page (which will be a likely scenario) the event handlers break.